### ISSUE TITLE
GDrive: Changes online do not get synced

### ISSUE THREAD

**Issue by loveisfoss** [hwittenborn/celeste#63]
I got a problem while trying to sync Google Drive. 

First sync works as expected, everything alright.

Later, I create a file in the cloud, that is located in 3 subfolders but Celeste does not sync it to local folder.
Likewise, if I change a previously existing document file in the same directory, it does not get synced either.

However, if I add or change a file on the cloud's root directory, it gets synced by celeste locally.

Any other experiencing something like this?

**Comment by Adi-ionut** [comment]
Yes, same here. If I make a change to let's say a text file locally, it won't sync in the cloud. It shows something like "directory not found".

Works the other way around. If I edit a file on the cloud it syncs locally also.

**Comment by loveisfoss** [comment]
> Yes, same here. If I make a change to let's say a text file locally, it won't sync in the cloud. It shows something like "directory not found".
> 
> Works the other way around. If I edit a file on the cloud it syncs locally also.

I think what you describe is another poblem. Yes, I can reproduce it also. If the file is deleted locally, I get the error "directory not found". Then, if I close Celeste and open it again, it passes the check of Gdrive directory without errors, but the file is still deleted locally while it continues to exist in the cloud.

My original problem is that **changes in the root GDrive directory are generally detected but changes in subfolders are not detected and the check progress passes without noticing them**.

After our discussion, I noticed that even changes in subfolders locally are not detected and thus, a new local file in a subfolder does not get uploaded to the cloud. Therefore, I change the issue title to better reflect the issue.

**Comment by hwittenborn** [comment]
I did some testing and I found the source of the issue - I was under the assumption that whenever a file was changed on a local system, that the timestamps for all parent directories got updated. Apparently this only happens to the direct parent folder, so that method probably won't work anymore. I'm gonna get #34 implemented as the workaround - if you have any thoughts about it feel free to send them, I'm gonna try to get that implemented soon though.

**Comment by hwittenborn** [comment]
I'm getting another change to get this looked at, and there doesn't seem to be a clean way to get this done currently. For the time being I think the approach I'm going to go with is to check a directory recursively for changes, and then just wait a few minutes before running sync checks again.

It's not the approach I'm wanting to go with, but the currently behavior is outright wrong and I need something that at least works for now. I'm still wanting to implement #34 as that's much more clean and efficient, but until then this will have to do.

**Comment by loveisfoss** [comment]
Nice, thanks for looking at it! I guess we can accept this workaround for now.

Just a thought: Since for now we work with looping the sync process, it would be nice to have a button to pause and resume it as we like.

**Comment by StevePires** [comment]
Hello and thanks for Celeste!

Is there any update on this matter?
I encountered the exact same issue (on Ubuntu 24.04.1) and ended up here

First Sync does sync everything nicely.
Then something in root folder seems to also sync correctly.
If something is moved in subfolder on the cloud, it disappears in local folder (seems to only be 'cut' from root folder but not 'paste' to destination).
Anything added in any local subfolder is not seen by the sync process (the process shows 'Checking [...] for changes' then 'files are synced' without actually syncing new files).

**Comment by Xanz** [comment]
Still an issue even coming into 2025 :(

Any files in sub directories do not sync at all after first sync, such as editing a file or creating a file in a sub directory. This is basically unusable, which is pretty disappointing.
